### PR TITLE
deps: update golangci-lint version used in CI

### PR DESCRIPTION
**Pull request by exdx**
Fixes the perma-failing sanity/lint action which is failing due to https://github.com/golangci/golangci-lint/issues/2374

### CODE REVIEW

File: .github/workflows/sanity.yaml [hunk 1]
```diff
@@ -24,4 +24,4 @@ jobs:
       - name: Run linting checks
         uses: "golangci/golangci-lint-action@v2"
         with:
-          version: "v1.43"
```
Comment (joelanford) [inline .github/workflows/sanity.yaml:27]: Can we update the version in the makefile as well to match?

Comment (joelanford) [reply .github/workflows/sanity.yaml:27]: Actually, I guess its `hack/tools/go.mod`

Comment (exdx) [reply .github/workflows/sanity.yaml:27]: Sure 👍 I updated it and ran `go mod tidy` on the tools go.mod as well

**Review by akihikokuroda — APPROVED** [review]
Thanks for fixing this.  Looks good to me!

**Review by timflannagan — APPROVED** [review]

**Review by joelanford — APPROVED** [review]

**Pull request merged**